packages/astro/src/core/build/static-build.ts

```diff
@@ -367,19 +362,27 @@ async function runPostBuildHooks(
 /**
  * For each statically prerendered page, replace their SSR file with a noop.
  * This allows us to run the SSR build only once, but still remove dependencies for statically rendered routes.
+ * If a component is shared between a statically rendered route and a SSR route, it will still be included in the SSR build.
  */
 async function cleanStaticOutput(
 	opts: StaticBuildOptions,
 	internals: BuildInternals,
 	ssrOutputChunkNames: string[]
 ) {
 	const allStaticFiles = new Set();
-	for (const pageData of eachPageData(internals)) {
-		if (pageData.route.prerender && !pageData.hasSharedModules) {
-			const { moduleSpecifier } = pageData;
-			const pageBundleId = internals.pageToBundleMap.get(moduleSpecifier);
-			const entryBundleId = internals.entrySpecifierToBundleMap.get(moduleSpecifier);
+	const allSSRFiles = new Set();
```

Comment:
I'd just rename this since the files added here are not necessarily "all SSR" like the "all static" above, they are not entirely static in opposition to the const above.

Maybe renaming them to `staticOnlyFiles` and `nonStaticOnlyFiles` would be more clear.

Reply:
In the docs, we use the term "on-demand". So maybe `onDemandFiles` is a better fit.